### ISSUE TITLE
Matrix entity positions have problems loading in-game.

### ISSUE THREAD

**Issue by Kneesnap** [Kneesnap/FrogLord#11]
If you move a positioned entity around, save and open in-game, in many scenarios the entity will just not show up, and effectively not be there. If you move it around, then move it back to the original position, the problem does not work.

It seems to me like there is some issue with position impacting some other value, possibly entity id or map groups, or position in entity list, etc. It's still unclear what the problem is, I may have to do some debugging with the engine to figure out the problem.

**Comment by Kneesnap** [comment]
Things to try:
 - If marking an entity as immortal bypasses the problem.
 - It seems like the entity problem has to do with positional sorting. It seems like the entity id is the problem, not the entity list order. I need to figure out which axes are weighted the most significant, or basically how the entity id is impacted.

**Comment by TehcJS** [comment]
I'll try those things. I've just got things set up so I'll try them soon.

**Comment by Kneesnap** [comment]
It appears this is likely only related to the position, nothing else.

I've narrowed it down to the specific part of the Frogger code which fails.
The problem is, I'm have a really difficult time figuring out how the value in the file is turned into the number the code uses. The value I leave in the file seems to not be the value the code gets, and I can't seem to figure out why this is. 

Another mystery is why the 16 lower bits seem to always be 1 or always be 0, with no discernable pattern why this is.

Even though this doesn't make sense, I managed to get positions to seemingly always work when X and Z are positive by doing this.

Shifting XYZ >> 16, when making the vector.

Editing with 4 bits fixed point decimal.

Shifting everything << 16 when it's time to put the position data back in the matrix.

Why this requires shifting by 16 is completely beyond me. I have to be missing something, since it just doesn't make sense to me.
